Frameworks/Foundation/NSDecimalNumber.mm

```diff
@@ -14,296 +14,344 @@
 //
 //******************************************************************************
 
-#import "Starboard.h"
-#import "StubReturn.h"
-#include <Foundation/Foundation.h>
+#import <Starboard.h>
+#import <StubReturn.h>
+#import <Foundation/Foundation.h>
+#import <Foundation/NSDecimal.h>
+#import <Foundation/NSDecimalNumberHandler.h>
+#import <vector>
+#import <algorithm>
+#import <limits>
 
 NSString* const NSDecimalNumberExactnessException = @"NSDecimalNumberExactnessException";
 NSString* const NSDecimalNumberOverflowException = @"NSDecimalNumberOverflowException";
 NSString* const NSDecimalNumberUnderflowException = @"NSDecimalNumberUnderflowException";
 NSString* const NSDecimalNumberDivideByZeroException = @"NSDecimalNumberDivideByZeroException";
 
+static const int _MAX_EXP = 127;
+
+// This is the limit of initWithDouble on the reference platform.
+static const double _DOUBLE_LIMIT = 1.8e146;
```

Comment:
>_DOUBLE_LIMIT [](start = 20, length = 13)

SHOUTY_NAMES are only allowed (and required) for preprocessor macros

[https://github.com/Microsoft/WinObjC/wiki/Coding-Standards](https://github.com/Microsoft/WinObjC/wiki/Coding-Standards)